internal/notifier/telegram.go

```diff
@@ -0,0 +1,68 @@
+package notifier
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/containrrr/shoutrrr"
+	"github.com/fluxcd/pkg/runtime/events"
+)
+
+type Telegram struct {
+	Channel string
+	Token   string
+}
+
+func NewTelegram(channel, token string) (*Telegram, error) {
+	if channel == "" {
+		return nil, errors.New("empty Telegram channel")
+	}
+
+	return &Telegram{
+		Channel: channel,
+		Token:   token,
+	}, nil
+}
+
+func (t *Telegram) Post(event events.Event) error {
+	// Skip any update events
+	if isCommitStatus(event.Metadata, "update") {
+		return nil
+	}
+
+	emoji := "💫"
+	if event.Severity == events.EventSeverityError {
+		emoji = "🚨"
+	}
+
+	message := fmt.Sprintf(`*%s %s/%s/%s*
+
+	%s`, emoji, strings.ToLower(event.InvolvedObject.Kind),
+		event.InvolvedObject.Name, event.InvolvedObject.Namespace, event.Message)
+	url := fmt.Sprintf("telegram://%s@telegram?channels=%s&parseMode=markDownv2", t.Token, t.Channel)
+	err := shoutrrr.Send(url, escapeString(message))
+	return err
+}
+
+// The telegram API requires that some special characters are escaped
+// in the message string. Docs: https://core.telegram.org/bots/api#formatting-options.
+// The function only handles a subset of the characters because the only special
+// characters in kubernetes names and namespaces are '-' and '.'.
```

Comment:
But `event.Message` can contain any character, for example a kubectl apply error.

Reply:
I will update the function

Reply:
Please test this with a Kustomization that fails to apply. I see that we don't post the revision but we should, check how alerts look for Slack: https://fluxcd.io/img/slack-error-alert.png

Reply:
We have to add the event metadata to the message body as separate field in the markdown, maybe a list?

Reply:
This is what the message looks like. With the metadata added

<img width="507" alt="Screenshot 2021-08-18 at 12 35 06 PM" src="https://user-images.githubusercontent.com/34774756/129891354-f9f61f46-e26a-4ce0-88c3-fa7f58d0ecd0.png">